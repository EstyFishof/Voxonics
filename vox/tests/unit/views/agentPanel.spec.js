import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import AgentPanel from '@/views/AgentPanel.vue';
import VueTippy, { TippyComponent } from "vue-tippy";

const localVue = createLocalVue();
localVue.use(Vuex);
const router = new VueRouter();
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);

describe('support', () => {
    window.env = {
        WS_URL: 'ws://168.119.92.172:9595'
    };
    const store = new Vuex.Store({
        actions: {
            pushLogout({commit}, value) {},
            emitError({commit}, value) {},
        },
    });

    const wrapper = shallowMount(AgentPanel, { localVue, router, store });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('logout toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'logout');
        await wrapper.vm.logout();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('computedTime toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'computedTime');
        await wrapper.vm.computedTime();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("online");
        expect(spy).toHaveReturnedWith({ color: '#00C82C', icon: wrapper.vm.OnlineIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("paused");
        expect(spy).toHaveReturnedWith({ color: '#FFB200', icon: wrapper.vm.PauseIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("offline");
        expect(spy).toHaveReturnedWith({ color: '#FF3A29', icon: wrapper.vm.OfflineIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("calling");
        expect(spy).toHaveReturnedWith({ color: '#78ABCE', icon: wrapper.vm.CallingIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("ringing");
        expect(spy).toHaveReturnedWith({ color: '#78ABCE', icon: wrapper.vm.CallingIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("talking");
        expect(spy).toHaveReturnedWith({ color: '#276899', icon: wrapper.vm.TalkingIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("blocked");
        expect(spy).toHaveReturnedWith({ color: '#636D75', icon: wrapper.vm.UnreachableIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor('');
        expect(spy).toHaveReturnedWith({ color: '#636D75', icon: wrapper.vm.UnreachableIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor('talking&calling');
        expect(spy).toHaveReturnedWith({ color: '#78ABCE', icon: wrapper.vm.CallingIcon });
        spy.mockRestore();
    });
    it('computed component', async () => {
        await AgentPanel.computed.elapsed.call(wrapper.vm);
    });
    it('mounted component', async () => {
        await AgentPanel.mounted.call(wrapper.vm);
    });
    it('beforeDestroy component', async () => {
        await AgentPanel.beforeDestroy.call(wrapper.vm);
    });
});