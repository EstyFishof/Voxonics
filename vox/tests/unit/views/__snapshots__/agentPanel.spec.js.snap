// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`support Check snapshot 1`] = `
<section
  class="content-wrapp"
>
  <section>
    <div
      class="container"
    >
      <div
        class="content"
      >
        <div
          class="title-container"
        >
          <div
            class="text ava"
          >
            <div
              class="avatar"
            >
              <img
                alt="user-avatar"
                src=""
              />
            </div>
            
            Hi, !
          
          </div>
        </div>
         
        <table
          class="user-table"
        >
          <thead>
            <tr
              class="table-header"
            >
              <th>
                Agent Number(ID)
              </th>
               
              <th>
                Calls Today
              </th>
               
              <th>
                Status
              </th>
               
              <th>
                Action
              </th>
            </tr>
          </thead>
           
          <tbody>
            <tr>
              <td
                class="text"
              >
                
              
            
              </td>
               
              <td
                class="text"
              />
               
              <td
                class="status"
                style="color: rgb(99, 109, 117);"
              >
                <span
                  class="days"
                  content="Days"
                  style="background: rgb(99, 109, 117);"
                  tabindex="0"
                >
                  <span
                    class="circle"
                  >
                    
                    00
                  
                  </span>
                </span>
                 
                <img
                  alt="phone"
                  class="status-phone"
                  src=""
                />
                
               00:00
            
              </td>
               
              <td>
                <div
                  class="btn-wrapper"
                >
                  <svg
                    class="btn disabled"
                    fill="none"
                    height="28"
                    viewBox="0 0 90 28"
                    width="90"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="27"
                      rx="1.5"
                      stroke="#00C82C"
                      width="89"
                      x="0.5"
                      y="0.5"
                    />
                     
                    <path
                      d="M41.1787 15.8232H38.1729V21H36.4766V8.20312H40.7129C42.1543 8.20312 43.2617 8.53125 44.0352 9.1875C44.8145 9.84375 45.2041 10.7988 45.2041 12.0527C45.2041 12.8496 44.9873 13.5439 44.5537 14.1357C44.126 14.7275 43.5283 15.1699 42.7607 15.4629L45.7666 20.8945V21H43.9561L41.1787 15.8232ZM38.1729 14.4434H40.7656C41.6035 14.4434 42.2686 14.2266 42.7607 13.793C43.2588 13.3594 43.5078 12.7793 43.5078 12.0527C43.5078 11.2617 43.2705 10.6553 42.7959 10.2334C42.3271 9.81152 41.6475 9.59766 40.7568 9.5918H38.1729V14.4434ZM51.2686 21.1758C49.9795 21.1758 48.9307 20.7539 48.1221 19.9102C47.3135 19.0605 46.9092 17.9268 46.9092 16.5088V16.21C46.9092 15.2666 47.0879 14.4258 47.4453 13.6875C47.8086 12.9434 48.3125 12.3633 48.957 11.9473C49.6074 11.5254 50.3105 11.3145 51.0664 11.3145C52.3027 11.3145 53.2637 11.7217 53.9492 12.5361C54.6348 13.3506 54.9775 14.5166 54.9775 16.0342V16.7109H48.5352C48.5586 17.6484 48.8311 18.4072 49.3525 18.9873C49.8799 19.5615 50.5479 19.8486 51.3564 19.8486C51.9307 19.8486 52.417 19.7314 52.8154 19.4971C53.2139 19.2627 53.5625 18.9521 53.8613 18.5654L54.8545 19.3389C54.0576 20.5635 52.8623 21.1758 51.2686 21.1758ZM51.0664 12.6504C50.4102 12.6504 49.8594 12.8906 49.4141 13.3711C48.9688 13.8457 48.6934 14.5137 48.5879 15.375H53.3516V15.252C53.3047 14.4258 53.082 13.7871 52.6836 13.3359C52.2852 12.8789 51.7461 12.6504 51.0664 12.6504ZM62.7383 21C62.6445 20.8125 62.5684 20.4785 62.5098 19.998C61.7539 20.7832 60.8516 21.1758 59.8027 21.1758C58.8652 21.1758 58.0947 20.9121 57.4912 20.3848C56.8936 19.8516 56.5947 19.1777 56.5947 18.3633C56.5947 17.373 56.9697 16.6055 57.7197 16.0605C58.4756 15.5098 59.5361 15.2344 60.9014 15.2344H62.4834V14.4873C62.4834 13.9189 62.3135 13.4678 61.9736 13.1338C61.6338 12.7939 61.1328 12.624 60.4707 12.624C59.8906 12.624 59.4043 12.7705 59.0117 13.0635C58.6191 13.3564 58.4229 13.7109 58.4229 14.127H56.7881C56.7881 13.6523 56.9551 13.1953 57.2891 12.7559C57.6289 12.3105 58.0859 11.959 58.6602 11.7012C59.2402 11.4434 59.876 11.3145 60.5674 11.3145C61.6631 11.3145 62.5215 11.5898 63.1426 12.1406C63.7637 12.6855 64.0859 13.4385 64.1094 14.3994V18.7764C64.1094 19.6494 64.2207 20.3438 64.4434 20.8594V21H62.7383ZM60.04 19.7607C60.5498 19.7607 61.0332 19.6289 61.4902 19.3652C61.9473 19.1016 62.2783 18.7588 62.4834 18.3369V16.3857H61.209C59.2168 16.3857 58.2207 16.9688 58.2207 18.1348C58.2207 18.6445 58.3906 19.043 58.7305 19.3301C59.0703 19.6172 59.5068 19.7607 60.04 19.7607ZM66.2627 16.166C66.2627 14.707 66.6084 13.5352 67.2998 12.6504C67.9912 11.7598 68.8965 11.3145 70.0156 11.3145C71.1289 11.3145 72.0107 11.6953 72.6611 12.457V7.5H74.2871V21H72.793L72.7139 19.9805C72.0635 20.7773 71.1582 21.1758 69.998 21.1758C68.8965 21.1758 67.9971 20.7246 67.2998 19.8223C66.6084 18.9199 66.2627 17.7422 66.2627 16.2891V16.166ZM67.8887 16.3506C67.8887 17.4287 68.1113 18.2725 68.5566 18.8818C69.002 19.4912 69.6172 19.7959 70.4023 19.7959C71.4336 19.7959 72.1865 19.333 72.6611 18.4072V14.0391C72.1748 13.1426 71.4277 12.6943 70.4199 12.6943C69.623 12.6943 69.002 13.002 68.5566 13.6172C68.1113 14.2324 67.8887 15.1436 67.8887 16.3506ZM79.9297 18.6182L82.1445 11.4902H83.8848L80.0615 22.4678C79.4697 24.0498 78.5293 24.8408 77.2402 24.8408L76.9326 24.8145L76.3262 24.7002V23.3818L76.7656 23.417C77.3164 23.417 77.7441 23.3057 78.0488 23.083C78.3594 22.8604 78.6143 22.4531 78.8135 21.8613L79.1738 20.8945L75.7812 11.4902H77.5566L79.9297 18.6182Z"
                      fill="#00C82C"
                    />
                     
                    <circle
                      cx="16"
                      cy="14"
                      r="10.5"
                      stroke="#00C82C"
                    />
                     
                    <path
                      class="accept"
                      d="M11 13.4758L15.037 17L23 10"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                     
                    <circle
                      class="subs"
                      cx="26.5"
                      cy="19.5"
                      r="6"
                      stroke="#FFB200"
                    />
                     
                    <rect
                      class="vertical-pause"
                      height="7"
                      rx="1.07143"
                      width="2.14286"
                      x="24"
                      y="16"
                    />
                     
                    <rect
                      class="vertical-pause"
                      height="7"
                      rx="1.07143"
                      width="2.14286"
                      x="27"
                      y="16"
                    />
                  </svg>
                   
                  <svg
                    class="btn disabled"
                    fill="none"
                    height="28"
                    style="display: none;"
                    viewBox="0 0 90 28"
                    width="90"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      class="subs"
                      height="27"
                      rx="1.5"
                      stroke="#FFB200"
                      width="89"
                      x="0.5"
                      y="0.5"
                    />
                     
                    <path
                      d="M38.1729 15.9902V21H36.4854V8.20312H41.2051C42.6055 8.20312 43.7012 8.56055 44.4922 9.27539C45.2891 9.99023 45.6875 10.9365 45.6875 12.1143C45.6875 13.3564 45.2979 14.3145 44.5186 14.9883C43.7451 15.6562 42.6348 15.9902 41.1875 15.9902H38.1729ZM38.1729 14.6104H41.2051C42.1074 14.6104 42.7988 14.3994 43.2793 13.9775C43.7598 13.5498 44 12.9346 44 12.1318C44 11.3701 43.7598 10.7607 43.2793 10.3037C42.7988 9.84668 42.1396 9.60938 41.3018 9.5918H38.1729V14.6104ZM53.3516 21C53.2578 20.8125 53.1816 20.4785 53.123 19.998C52.3672 20.7832 51.4648 21.1758 50.416 21.1758C49.4785 21.1758 48.708 20.9121 48.1045 20.3848C47.5068 19.8516 47.208 19.1777 47.208 18.3633C47.208 17.373 47.583 16.6055 48.333 16.0605C49.0889 15.5098 50.1494 15.2344 51.5146 15.2344H53.0967V14.4873C53.0967 13.9189 52.9268 13.4678 52.5869 13.1338C52.2471 12.7939 51.7461 12.624 51.084 12.624C50.5039 12.624 50.0176 12.7705 49.625 13.0635C49.2324 13.3564 49.0361 13.7109 49.0361 14.127H47.4014C47.4014 13.6523 47.5684 13.1953 47.9023 12.7559C48.2422 12.3105 48.6992 11.959 49.2734 11.7012C49.8535 11.4434 50.4893 11.3145 51.1807 11.3145C52.2764 11.3145 53.1348 11.5898 53.7559 12.1406C54.377 12.6855 54.6992 13.4385 54.7227 14.3994V18.7764C54.7227 19.6494 54.834 20.3438 55.0566 20.8594V21H53.3516ZM50.6533 19.7607C51.1631 19.7607 51.6465 19.6289 52.1035 19.3652C52.5605 19.1016 52.8916 18.7588 53.0967 18.3369V16.3857H51.8223C49.8301 16.3857 48.834 16.9688 48.834 18.1348C48.834 18.6445 49.0039 19.043 49.3438 19.3301C49.6836 19.6172 50.1201 19.7607 50.6533 19.7607ZM63.1426 20.0596C62.5098 20.8037 61.5811 21.1758 60.3564 21.1758C59.3428 21.1758 58.5693 20.8828 58.0361 20.2969C57.5088 19.7051 57.2422 18.832 57.2363 17.6777V11.4902H58.8623V17.6338C58.8623 19.0752 59.4482 19.7959 60.6201 19.7959C61.8623 19.7959 62.6885 19.333 63.0986 18.4072V11.4902H64.7246V21H63.1777L63.1426 20.0596ZM72.7402 18.4775C72.7402 18.0381 72.5732 17.6982 72.2393 17.458C71.9111 17.2119 71.334 17.001 70.5078 16.8252C69.6875 16.6494 69.0342 16.4385 68.5479 16.1924C68.0674 15.9463 67.71 15.6533 67.4756 15.3135C67.2471 14.9736 67.1328 14.5693 67.1328 14.1006C67.1328 13.3213 67.4609 12.6621 68.1172 12.123C68.7793 11.584 69.623 11.3145 70.6484 11.3145C71.7266 11.3145 72.5996 11.5928 73.2676 12.1494C73.9414 12.7061 74.2783 13.418 74.2783 14.2852H72.6436C72.6436 13.8398 72.4531 13.4561 72.0723 13.1338C71.6973 12.8115 71.2227 12.6504 70.6484 12.6504C70.0566 12.6504 69.5938 12.7793 69.2598 13.0371C68.9258 13.2949 68.7588 13.6318 68.7588 14.0479C68.7588 14.4404 68.9141 14.7363 69.2246 14.9355C69.5352 15.1348 70.0947 15.3252 70.9033 15.5068C71.7178 15.6885 72.377 15.9053 72.8809 16.1572C73.3848 16.4092 73.7568 16.7139 73.9971 17.0713C74.2432 17.4229 74.3662 17.8535 74.3662 18.3633C74.3662 19.2129 74.0264 19.8955 73.3467 20.4111C72.667 20.9209 71.7852 21.1758 70.7012 21.1758C69.9395 21.1758 69.2656 21.041 68.6797 20.7715C68.0938 20.502 67.6338 20.127 67.2998 19.6465C66.9717 19.1602 66.8076 18.6357 66.8076 18.0732H68.4336C68.4629 18.6182 68.6797 19.0518 69.084 19.374C69.4941 19.6904 70.0332 19.8486 70.7012 19.8486C71.3164 19.8486 71.8086 19.7256 72.1777 19.4795C72.5527 19.2275 72.7402 18.8936 72.7402 18.4775ZM80.4307 21.1758C79.1416 21.1758 78.0928 20.7539 77.2842 19.9102C76.4756 19.0605 76.0713 17.9268 76.0713 16.5088V16.21C76.0713 15.2666 76.25 14.4258 76.6074 13.6875C76.9707 12.9434 77.4746 12.3633 78.1191 11.9473C78.7695 11.5254 79.4727 11.3145 80.2285 11.3145C81.4648 11.3145 82.4258 11.7217 83.1113 12.5361C83.7969 13.3506 84.1396 14.5166 84.1396 16.0342V16.7109H77.6973C77.7207 17.6484 77.9932 18.4072 78.5146 18.9873C79.042 19.5615 79.71 19.8486 80.5186 19.8486C81.0928 19.8486 81.5791 19.7314 81.9775 19.4971C82.376 19.2627 82.7246 18.9521 83.0234 18.5654L84.0166 19.3389C83.2197 20.5635 82.0244 21.1758 80.4307 21.1758ZM80.2285 12.6504C79.5723 12.6504 79.0215 12.8906 78.5762 13.3711C78.1309 13.8457 77.8555 14.5137 77.75 15.375H82.5137V15.252C82.4668 14.4258 82.2441 13.7871 81.8457 13.3359C81.4473 12.8789 80.9082 12.6504 80.2285 12.6504Z"
                      fill="#FFB200"
                    />
                     
                    <circle
                      cx="16"
                      cy="14"
                      r="10.5"
                      stroke="#FFB200"
                    />
                     
                    <circle
                      class="subs"
                      cx="26.5"
                      cy="19.5"
                      r="6"
                      stroke="#00C82C"
                    />
                     
                    <path
                      class="accept"
                      d="M24 19.9862L26.0185 22L30 18"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                     
                    <rect
                      class="vertical-pause"
                      height="10"
                      rx="1.53061"
                      width="3.06122"
                      x="12"
                      y="9"
                    />
                     
                    <rect
                      class="vertical-pause"
                      height="10"
                      rx="1.53061"
                      width="3.06122"
                      x="17"
                      y="9"
                    />
                  </svg>
                   
                  <span
                    class="item-menu-btn menu-log-out"
                  >
                    <img
                      class="log-out-icon"
                      src="@/assets/img/icons/menu-logout-agentsMap.svg"
                    />
                    
                    Log out
                  
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </section>
</section>
`;
